File: backend/models/part.py
from sqlalchemy import Column, Integer, String, Float, Boolean, ForeignKey
from sqlalchemy.orm import relationship
from backend.database.base import Base


class Part(Base):
    __tablename__ = 'parts'

    id = Column(Integer, primary_key=True, index=True)
    claim_id = Column(Integer, ForeignKey('claims.id'), nullable=False)
    part_name = Column(String, nullable=False)
    quantity = Column(Integer, default=1)
    rate = Column(Float, default=0.0)
    amount = Column(Float, default=0.0)
    material_type = Column(String, default='')
    depreciation_percent = Column(Float, default=0.0)
    approved_amount = Column(Float, default=0.0)
    accepted = Column(Boolean, default=False)

    claim = relationship('Claim', back_populates='parts')

File: lib/models/part.dart
class PartItem {
  final int id;
  final int claimId;
  String partName;
  int quantity;
  double rate;
  double amount;
  String materialType;
  double depreciationPercent;
  double approvedAmount;
  bool accepted;

  PartItem({
    required this.id,
    required this.claimId,
    required this.partName,
    required this.quantity,
    required this.rate,
    required this.amount,
    required this.materialType,
    this.depreciationPercent = 0.0,
    this.approvedAmount = 0.0,
    this.accepted = false,
  });

  factory PartItem.fromJson(Map<String, dynamic> json) => PartItem(
        id: json['id'] as int? ?? 0,
        claimId: json['claim_id'] as int? ?? 0,
        partName: json['part_name'] as String? ?? '',
        quantity: json['quantity'] as int? ?? 1,
        rate: (json['rate'] as num?)?.toDouble() ?? 0.0,
        amount: (json['amount'] as num?)?.toDouble() ?? 0.0,
        materialType: json['material_type'] as String? ?? '',
        depreciationPercent:
            (json['depreciation_percent'] as num?)?.toDouble() ?? 0.0,
        approvedAmount:
            (json['approved_amount'] as num?)?.toDouble() ?? 0.0,
        accepted: json['accepted'] as bool? ?? false,
      );

  Map<String, dynamic> toJson() => {
        'id': id,
        'claim_id': claimId,
        'part_name': partName,
        'quantity': quantity,
        'rate': rate,
        'amount': amount,
        'material_type': materialType,
        'depreciation_percent': depreciationPercent,
        'approved_amount': approvedAmount,
        'accepted': accepted,
      };
}
